generate/generate_nircam.py

```diff
@@ -318,100 +342,126 @@
-compare.compare_siaf(new_siaf, reference_siaf_input=ref_siaf, fractional_tolerance=1e-6, ignore_attributes=ignore_attributes)
-# compare.compare_siaf(new_siaf, reference_siaf_input=ref_siaf, fractional_tolerance=1e-4, ignore_attributes=ignore_attributes, report_dir=test_dir)
-# compare.compare_siaf(new_siaf, reference_siaf_input=ref_siaf, fractional_tolerance=1e-6)
+    # write the SIAFXML to disk
+    filenames = pysiaf.iando.write.write_jwst_siaf(aperture_collection, basepath=test_dir, file_format=['xml'], label='pysiaf')
+    print('SIAFXML written in {}'.format(filenames[0]))
 
+    # compare to SIAFXML produced the old way
+    # ref_siaf = pysiaf.Siaf(instrument, os.path.join(test_dir , '{}'.format('NIRCam_SIAF_2017-12-01.xml')))
+    ref_siaf = pysiaf.Siaf(instrument)
 
-roundtrip_table = compare.compare_transformation_roundtrip(new_siaf, reference_siaf_input=ref_siaf, report_dir=test_dir)
-# roundtrip_table = compare.compare_transformation_roundtrip(new_siaf, reference_siaf_input=ref_siaf, selected_aperture_name='NRCA5_FULL'.split(), report_dir=test_dir, make_plot=True)
-# roundtrip_table = compare.compare_transformation_roundtrip(new_siaf, reference_siaf_input=ref_siaf, selected_aperture_name=[s for s in aperture_name_list if 'FULL' in s], report_dir=test_dir, make_plot=False)
-
-1/0
-
-# illustrate fix of inverse NIRCam coefficients
-if 0:
-    import pylab as pl
-
-    pl.close('all')
+    if 0:
+        pre_delivery_dir = os.path.join(JWST_DELIVERY_DATA_ROOT, instrument)
+        pre_delivery_siaf = pysiaf.Siaf(instrument, basepath=pre_delivery_dir)
+        ref_siaf = pre_delivery_siaf
 
-    fig = pl.figure(figsize=(12, 6), facecolor='w', edgecolor='k')
-    pl.clf()
-    # pl.subplot(2,1,1)
-    pl.plot(roundtrip_table['siaf0_dx_mean'], 'b-', label='PRD dx_mean')
-    pl.plot(roundtrip_table['siaf0_dy_mean'], 'r-', label='PRD dy_mean')
-    pl.plot(roundtrip_table['siaf1_dx_mean'], 'ko--', label='fixed dx_mean')
-    pl.plot(roundtrip_table['siaf1_dy_mean'], 'go--', label='fixed dy_mean')
-    pl.title('Mean absolute difference')
-    pl.legend()
-    # pl.subplot(2,1,2)
-    # pl.plot(roundtrip_table['siaf0_dx_rms'], 'b-', label='PRD dx_mean')
-    # pl.plot(roundtrip_table['siaf0_dy_rms'], 'r-', label='PRD dy_mean')
-    # pl.plot(roundtrip_table['siaf1_dx_rms'], 'ko--', label='fixed dx_mean')
-    # pl.plot(roundtrip_table['siaf1_dy_rms'], 'go--', label='fixed dy_mean')
-    # pl.title('RMS absolute difference')
+    new_siaf = pysiaf.Siaf(instrument, filenames[0])
 
-    pl.xticks(np.arange(len(roundtrip_table)), roundtrip_table['AperName'], rotation='vertical')
-    pl.margins(0.2)
-    pl.subplots_adjust(bottom=0.15)
-    pl.show()
+    # compare.compare_siaf(new_siaf, reference_siaf_input=ref_siaf, fractional_tolerance=1e-1, selected_aperture_name=master_aperture_names)#['NRCA3_FULL_OSS', 'NRCA1_FULL_OSS']) # 'NRCA4_SUB160', 'NRCA4_FULL', 'NRCA3_SUB160', 'NRCA3_FULL', 'NRCA5_SUB400P', 'NRCB5_SUB400P',
```

Comment:
Could all the compare checks (lines 413-416 and 421-422, 426-427) be moved somewhere else (like a separate function) or just deleted?